functions/Restore-DbaBackup.ps1

```diff
@@ -0,0 +1,215 @@
+The SQL Server instance. 
+
+.PARAMETER SqlCredential
+Allows you to login to servers using SQL Logins as opposed to Windows Auth/Integrated/Trusted. 
+
+.PARAMETER Path
+Path to SQL Server backup file. This can be a full, differential or log backup file.
+
+.PARAMETER Files
+A Files object(s) containing SQL Server backup files. 
+
+.PARAMETER RestoreLocation
+Path to restore the SQL Server backups to on the target inance
+
+.PARAMETER RestoreTime
+Specify a DateTime object to which you want the database restored to. Default is to the latest point available 
+
+.PARAMETER OlaStyle
+Switch to indicate the backup files are in a folder structure as created by Ola Hallengreen's maintenance scripts
+
+.PARAMETER FileDirectory
+Switch to indicate the backup files just exist in a folder (this is the standard)
+	
+.PARAMETER FileList
+Returns detailed information about the files within the backup
+
+.PARAMETER DatabaseName
+Name to restore the database under
+
+.PARAMETER NoRecovery
+Indicates if the database should be recovered after last restore. Default is to recover
+
+.PARAMETER ReplaceDatabase
+Switch indicated is the restore is allowed to replace an existing database.
+
+.PARAMETER Scripts
+Switch to indicate if T-SQL restore scripts should be written out
+
+.PARAMETER ScriptOnly
+Switch indicates that ONLY T-SQL scripts should be generated, no restore takes place
+
+.PARAMETER VerifyOnly
+Switch indicate that restore should be verified
+
+.PARAMETER XpDirTree
+Switch that indicated file scanning should be performed by the SQL Server instance using xp_dirtree
+You must have sysadmin role membership on the instance for this to work.
+
+.PARAMETER LogicalFileMapping
+Accepts a hashtable of mappings to be used to rename Logical Files post restore, of the form:
+$mapping = @{'LogicalFile1'='newname1';'LogicalFile2'='othername'}
+You don't have to specify all files, and just those mapped will  be renamed
+Exclusive with LogicalFilePrefix
+
+.PARAMETER LogicalFilePrefix
+Specify a string which will be prefixed to ALL logical files post restore.
+Exlusive with LogicalFileMapping
+.NOTES
+Original Author: Stuart Moore (@napalmgram), stuart-moore.com
+
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.EXAMPLE
+Restore-DbaBackup -SqlServer server1\instance1 -path \\server2\backups\$ 
```

Comment:
Ah, that's from my copied examples. we tend to 'hide' shares at work with a $ on the end. I'll tidy them away